src/main/java/net/dries007/tfc/compat/jade/common/BlockEntityTooltips.java

```diff
@@ -0,0 +1,332 @@
+                {
+                    tooltip.accept(Helpers.translatable("tfc.tooltip.pot_boiling"));
+                }
+                else if (pot.getOutput() != null && !pot.getOutput().isEmpty())
+                {
+                    tooltip.accept(Helpers.translatable("tfc.tooltip.pot_finished"));
+
+                    if (pot.getOutput() instanceof SoupPotRecipe.SoupOutput soup)
+                    {
+                        final ItemStack stack = soup.stack();
+                        itemWithCount(tooltip, stack);
+
+                        final List<Component> text = new ArrayList<>();
+                        stack.getCapability(FoodCapability.CAPABILITY).ifPresent(cap -> cap.addTooltipInfo(stack, text));
+                        text.forEach(tooltip);
+                    }
+                }
+            }
+        }
+    };
+
+    public static final BlockEntityTooltip FRUIT_TREE_SAPLING = (level, state, entity, tooltip) -> {
+        if (entity instanceof TickCounterBlockEntity counter && state.getBlock() instanceof FruitTreeSaplingBlock sapling)
+        {
+            final String growth = Helpers.formatPercentage(Math.min(0.99f, (float) counter.getTicksSinceUpdate() / (sapling.getTreeGrowthDays() * ICalendar.TICKS_IN_DAY)) * 100);
+            tooltip.accept(Helpers.translatable("tfc.jade.growth").append(growth));
+            hoeOverlay(level, sapling, entity, tooltip);
+        }
+    };
+
+    public static final BlockEntityTooltip HOE_OVERLAY = (level, state, entity, tooltip) -> {
+        if (state.getBlock() instanceof HoeOverlayBlock overlay)
+        {
+            hoeOverlay(level, overlay, entity, tooltip);
+        }
+    };
+
+    public static final BlockEntityTooltip LAMP = (level, state, entity, tooltip) -> {
+        if (entity instanceof LampBlockEntity lamp && state.getBlock() instanceof LampBlock)
+        {
+            // todo: this info is not available on the client (?)
+            LampFuel fuel = lamp.getFuel();
+            if (fuel != null)
+            {
+                tooltip.accept(Helpers.translatable("tfc.jade.burn_rate", String.valueOf(fuel.getBurnRate())));
+                if (state.getValue(LampBlock.LIT))
+                {
+                    if (fuel.getBurnRate() == -1)
+                    {
+                        tooltip.accept(Helpers.translatable("tfc.jade.burn_forever"));
+                    }
+                    else
+                    {
+                        lamp.getCapability(Capabilities.FLUID).ifPresent(cap -> {
+                            final int fluid = cap.getFluidInTank(0).getAmount();
+                            if (fluid > 0)
+                            {
+                                // ticks / mB * mB = ticks
+                                tooltip.accept(Helpers.translatable("tfc.jade.hours_remaining", String.valueOf(Math.round((float) fluid * fuel.getBurnRate() / ICalendar.TICKS_IN_HOUR))));
```

Comment:
Use `Calendars.get(level).getTimeDelta(fluid * fuel.getBurnRate())`

Standardized tooltips!!

You might still want a `Remaining: %s` tooltip and then use the time delate as the `%s` key.